src/config/templates.toml

```diff
@@ -10,10 +10,13 @@ log = '''
 label(if(current_working_copy, "working_copy"),
   concat(
     separate(" ",
-      if(divergent,
-        label("divergent", format_short_change_id(change_id) ++ "??"),
-        if(obsolete,
-          label("obsolete", format_short_change_id(change_id) ++ " hidden"),
+      if(obsolete,
+        label("obsolete",
+          (format_short_change_id(change_id) ++
+            if(divergent, label("divergent", "??"), "") ++
+            " hidden")),
+        if(divergent,
+          label("divergent", format_short_change_id(change_id) ++ "??"),
```

Comment:
I don't follow closely, but maybe you can build labels and contents separately:
```
label(
  separate(" ", if(divergent, "divergent"), if(obsolete, "obsolete")),
  format_short_change_id(change_id) ++ if(divergent, "??") ...
```

Reply:
I didn't realize you can specify more than one label at a time. I'll try this, will need to update the colors accordingly.

Reply:
Yes, I think this works better. Now colors.toml is more complex, but that's fine.